crmd/callbacks.c

```diff
@@ -217,9 +217,9 @@ peer_update_callback(enum crm_status_type type, crm_node_t * node, const void *d
                 crm_trace("Updating CIB %s stonithd reported fencing of %s complete",
                           (down->confirmed? "after" : "before"), node->uname);
 
-            } else if (alive == FALSE) {
+            } else if ((alive == FALSE) && safe_str_eq(task, CRM_OP_SHUTDOWN)) {
                 crm_notice("%s of %s (op %d) is complete", task, node->uname, down->id);
-                /* down->confirmed = TRUE; Only stonith-ng returning should imply completion */
+                /* down->confirmed = TRUE; */
```

Comment:
This technique (or the similar "#if 0") is often used to draw attention to an alternative approach that was considered and rejected. If someone is working on that section of code, it might be useful for them to think through the implications of that approach, either to understand why it was rejected, or to consider re-enabling it if potential concerns can be addressed. git blame can be used to find the commit that commented it, which will sometimes have more details.